### ISSUE TITLE
Handling perl5.16 on 10.13

### ISSUE THREAD

**Issue by dmacks** [fink/fink-distributions#6]
Apple has perl-5.16.2 on 10.9 and perl-5.18.2 on 10.10 and beyond. We supply "the other" of those two versions on 10.9-10.12 as our usual pattern of backward compatibility for all OS X that the dist supports. However, we do not currently have perl-5.16.2 on 10.13. However, we do have all the -pm5162 variants of modules on 10.13. Either we need to bring the interpretter up to support the modules, or stop bringing the modules along.

The only complication I can see is texi2html, which hardcodes perl5.16 because it relies on a deterministic ordering of a certain datastructure that is not inherently ordered but was ordered in perl up through 5.16. I don't know if the now-random ordering just causes test failures because some expected data isn't in the same order, or if the actual behavior is now non-deterministic.

**Comment by nieder** [comment]
For the record, Debian/Ubuntu and OpenSUSE have patches to texi2html.pl that fix sort randomness. They do NOT fix the test failure when I tested against /usr/bin/perl5.18 on 10.11.

https://packages.ubuntu.com/source/artful/texi2html
https://www.mail-archive.com/debian-bugs-dist@lists.debian.org/msg1344442.html
https://build.opensuse.org/package/view_file/Publishing/texi2html/texi2html-5584.patch?expand=1
https://wiki.debian.org/ReproducibleBuilds/RandomnessInHTMLGeneratedByTexi2html

I only tried the `PERL_HASH_SEED=0` fix by setting it as an EVNVAR in TestScript. Maybe setting inside the script that calls texi2html.pl during TestScript might yield better results (assuming it deals with the same sort of ordering).

**Comment by dmacks** [comment]
I looked at the texi2html test failures, and some seem to be entirely different values (html anchors of form \d vs \d-\d) not just different orders of certain chunks of output. The debian and/or opensuse patches did not fix the results. I'm wondering if we should just haul a private perl5.16 into that package and get on with the killing-5.16.2 process. Or at least turn off the *-pm5162 modules and keep just perl5162-core for legacy support.

**Comment by dmacks** [comment]
texi2html is actually already broken on 10.13 because it hardcodes /usr/bin/perl5.16 in the runtime (not just test-suite), which only exists up through 10.12. This is a blocker in the bindist and for some users.

There is also nondeterministic .deb results on 10.9 vs 10.{10,11,12} because 10.9 uses /usr/bin/perl in that context rather than /usr/bin/perl5.16 (/usr/bin/perl became 5.18 as of 10.10). I assume 10.9 could use that versioned one for consistency rather than having a separate higher-revision package for 10.10+, but we still need a fink perl5.16 (or to fix texi2html for 5.18) in order to have a viable texi2html at all beyond 10.12.

And there is also-also inconsistent results for the USE_UNIDECODE value hardcoded into the runtime (even in the bindist, so it's not just a missing dependency?): it's 1 on 10.{9,10} but 0 on 10.{11,12}.

. The ./configure test is:
```
     if $PERL -e "use Encode; use Unicode::Normalize;" > /dev/null 2>&1; 
     then USE_UNICODE=1
     else
         USE_UNICODE=0
     fi
[...]
     USE_UNIDECODE=0
     if test $USE_UNICODE = 1; then
        if $PERL -e "use Text::Unidecode;" > /dev/null 2>&1; 
        then USE_UNIDECODE=1
        fi
     fi
```
but Encode and Unicode::Normalize are part of perl core on all platforms and texi2html:Depends:texinfo:Depends:text-unidecode-pm, so I'm not sure how that result varies.

**Comment by dmacks** [comment]
I gave up on getting texi2html to work with 5.18, and pushed fink's perl5162 on 10.13. This has been a blocker for too many months already.

**Comment by akhansen** [comment]
Shall we close this?

**Comment by dmacks** [comment]
@Neider is it worth adding any of the patches you mentioned on Dec 14, 2017, even if they don't resolve the perl5162 requirement?

**Comment by nieder** [comment]
No idea, sorry.

**Comment by nieder** [comment]
I'm closing this.
perl5.16 is available through 10.14, explicitly not available on 10.15, and texi2html has it's own private perl on 10.15.
